src/core/mesh/qgsmeshlayer.h

```diff
@@ -289,6 +291,37 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
       */
     QgsMesh3dDataBlock dataset3dValue( const QgsMeshDatasetIndex &index, const QgsPointXY &point ) const;
 
+    /**
+      * Returns the value of 1D mesh dataset defined on edge that are in the rectangle and the closest of the center of this rectangle
+      *
+      * \note It uses previously cached and indexed triangular mesh
+      * and so if the layer has not been rendered previously
+      * (e.g. when used in a script) it returns NaN value
+      *
+      * \param index dataset index specifying group and dataset to extract value from
+      * \param rectangle the area where to search the edge, closest edge from center point will be used
+      * \returns interpolated value at the projected point. Returns NaN values for values
+      * outside the mesh layer and in case triangular mesh was not previously used for rendering
+      *
+      * \since QGIS 3.14
+      */
+    QgsMeshDatasetValue dataset1DValue( const QgsMeshDatasetIndex &index, const  QgsRectangle &rectangle ) const;
```

Comment:
use `double searchRadius` (no default value) to be consistent with `datasetValue`. Create rectangle internally.